### ISSUE TITLE
Models not found in log when starting XL

### ISSUE THREAD

**Issue by Darzoom** [smeighan/xLights#3771]
NOTE: IF YOU DO NOT ATTACH A SEQUENCE THAT DEMONSTRATES YOUR PROBLEM THEN THERE IS A HIGH PROBABILITY YOUR ISSUE WILL JUST BE CLOSED AND NOT INVESTIGATED.

**Describe the bug**
This is a multi part bug.
1.  When starting XL first time the log indicates the models can't be found.
2. When opening a previously completed sequence the timing marks are missing.  
3. When creating a new sequence and adding timing marks, saving the file, closing the sequence and reopening, the timing marks are missing.  

**To Reproduce**
(and/or link to a short video showing the problem with audio track describing what you are doing)
1. Launce XL and see if the models are found in the logs.
2. Create a new sequence and add timing  (beats) .
3. Save the sequence.
4. CLose the sequence
5. Reopen the sequence and the timing marks are missing. 

**Expected behavior**
A clear and concise description of what you expected to happen and why.

**Versions (please complete the following information):**
 - OS: [Mac Monterey 12.6.4]
 - xLights version - Occured in both 2023.4 and 2023.5.1 

**Attachments**

[Issue.zip](https://github.com/smeighan/xLights/files/11133661/Issue.zip)
[xLightsProblem.zip](https://github.com/smeighan/xLights/files/11133662/xLightsProblem.zip)
[xLights_dbgrpt-9782-20230402T192146.zip](https://github.com/smeighan/xLights/files/11133663/xLights_dbgrpt-9782-20230402T192146.zip)

**Comment by dkulp** [comment]
The timing marks "missing" is due to da4a98faecdb6a35fd01f5c36064fbea8db72fcd (and thus has been this way since 2022.24).    For some reason, your timing marks are 0 length which is why they aren't being loaded.    How are you creating them that they are 0 length?  We likely need to make sure in the UI that effects cannot be shrunk down to 0 length as well.

**Comment by Darzoom** [comment]
I am creating the timing marks with Queen Mary Vamp using beats and onsets and tried with multiple settings with the same results.

**Comment by Darzoom** [comment]
What about the warning of the missing models?

**Comment by dkulp** [comment]
I cannot reproduce this at all.   If I create a musical sequence and use the QM plugins to create timing tracks, I get proper timing tracks.  You may want to re-install the QM plugins to make sure you have the latest version.   
<img width="668" alt="Screenshot 2023-04-03 at 9 10 49 AM" src="https://user-images.githubusercontent.com/91569/229519831-070fb105-9bb1-4b87-8a9b-10fa63c1f7ef.png">

**Comment by Darzoom** [comment]
I reinstalled QM and same result.

**Comment by computergeek1507** [comment]
> I cannot reproduce this at all.   If I create a musical sequence and use the QM plugins to create timing tracks, I get proper timing tracks.  You may want to re-install the QM plugins to make sure you have the latest version.   
> <img width="668" alt="Screenshot 2023-04-03 at 9 10 49 AM" src="https://user-images.githubusercontent.com/91569/229519831-070fb105-9bb1-4b87-8a9b-10fa63c1f7ef.png">
> 

Are you running from the sandbox?

**Comment by Darzoom** [comment]
I don’t believe I am.  I can make a video clip of what I am doing or open a zoom session and show you.

**Comment by Darzoom** [comment]
How can the Aubio (Advance version of QMV) timing makes be made to work?

**Comment by derwin12** [comment]
These timing marks come from using the Advanced Queen Mary plugin - as noted these were valid until the change done v24.
So now we have sequences that will no longer work going forward. Can this be reverted or perhaps changed to allow the 0 length timing marks?

https://github.com/smeighan/xLights/commit/da4a98faecdb6a35fd01f5c36064fbea8db72fcd

![image](https://user-images.githubusercontent.com/4643499/234110500-f62e8383-0ed0-43cf-b4d3-f1a387d8d869.png)

**Comment by derwin12** [comment]
Is it case of adding this if condition here?
![image](https://user-images.githubusercontent.com/4643499/234113289-4297fb9a-daa1-486c-a460-a323a6093647.png)

**Comment by keithsw1111** [comment]
Timing tracks should never have zero length or overlapping timing marks. Our QM code previously allowed it and it shouldnt have. Deleting them is the right thing to do as they cause other issues.
That said the underlying problem has not been addressed but I have pushed a fix which will handle such plugins I hope better ensuring that *most* of the desired marks are present while at the same time ensuring the timing track follows the rules.
There are still some gap scenarios with the worst of the plugins (of which polyphonic transcription is one). If you want to use that one I suggest you instead import notes as it uses a slightly different approach which does even better at preserving all marks. That said the approach I implemented seems to handle most scenarios pretty well without the complexity of the note import code.
For those with offending timing tracks you are going to have to delete and re-add the timing track using the new code to hopefully restore most of the behaviour.
